python/ray/_raylet.pyx

```diff
@@ -3047,6 +3054,11 @@ cdef class CoreWorker:
         self._task_id_to_future = {}
         self.event_loop_executor = None
 
+        if RayConfig.instance().start_python_gc_manager_thread():
+            init_gc_collect_manager(ray_constants.RAY_GC_MIN_COLLECT_INTERVAL)
```

Comment:
let's make the thread an attribute of the core worker (see my other comment)

we also need to make sure to stop & join this thread when the core worker is shut down (for example if the user calls `ray.shutdown` explicitly, the thread should be stopped & joined)